internal/testutil/nopcloser.go

```diff
@@ -0,0 +1,19 @@
+package testutil
+
+import (
+	"io"
+)
+
+// readSeekNopCloser is an io.Reader that does nothing on Close, and
+// seeks to the beginning of the stream on Seek.
```

Comment:
> and seeks to the beginning of the stream on Seek

This is actually not true, right? It can seek to wherever in the file because it satisfies the interface. And `is an io.Reader` is also not true because it is actually an `io.ReadSeeker`.

I would actually drop this comment and only keep:
```go
// readSeekNopCloser is an io.ReadSeeker that does nothing on Close.
type ...

// It is an extension of io.NopCloser that also implements io.Seeker.
func ReadSeekNopCloser ...
```